### DISCUSSION TITLE
Compatibility with ReSeed Cartridge

### DISCUSSION THREAD

**Discussion by SteveGuidi** [frntc/SIDKick-pico#36] [General]
Linking to the question I asked about ReSeed support on the SIDKick (Teensy) project. 

https://github.com/frntc/SIDKick/discussions/4#discussion-6469138

It would be great to know if we can use the Pico module with ReSeed as well!

**Comment by frntc** [comment]
(almost the same answer as for the SIDKick)

SukkoPera kindly sent me a ReSeed cartridge. I'll adapt the firmware once I find the time to set up my C16 :-) (it will then go together with a new firmware release as there will be quite a few changes).

The configuration menu, however, will require more work, not sure yet how I will do this.

**Reply by SteveGuidi** [reply]
I've been giving this some thought this past week as I have a SIDKick-Pico that I'm eager to assemble and try out.  I think that the main challenge is to ensure that the SIDKick reads appropriate input pin signals at the correct clock frequency.  ReSeed uses the $FD40-$FD5F address range, and software typically copies whatever is in $D400-$D41F to the reserved address for C64 music playback. Who knows? It might already do this -- I'll test and report back.  (FWIW, ArmSid works but BackSid does not)

Regarding the config menu, I'm not sure how to achieve the same results as the C64 version apart from porting the code and bootstrapping to an address in the $FD40-$FD5F range. (i.e. SYS 54301` becomes SYS 64861)  I'm just thinking out loud here -- I don't really know how this works.to begin with 😁!

I'm happy to help troubleshoot further and try different builds or tweaks as needed.

**Reply by frntc** [reply]
ReSeed takes care that the address range is mapped to the SID-socket (via the CS-signal). However, the C64/C128 firmware will not work out of the box. I have almost finished the C16/+4 firmware and config tool, but didn't find the time yet to do the remaining bits.

**Reply by SteveGuidi** [reply]
This is fantastic -- thank you for spending the time to enable ReSeed compatibility! Very eager to try things out when ready!
